sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -267,13 +274,27 @@ export class ServiceBusManagementClient extends ServiceClient {
     operationOptions?: OperationOptions
   ): Promise<NamespacePropertiesResponse> {
     log.httpAtomXml(`Performing management operation - getNamespaceProperties()`);
-    const response: HttpOperationResponse = await this.getResource(
-      "$namespaceinfo",
-      this.namespaceResourceSerializer,
+    const { span, updatedOperationOptions } = createSpan(
+      "ServiceBusManagementClient-getNamespaceProperties",
       operationOptions
     );
+    try {
+      const response: HttpOperationResponse = await this.getResource(
+        "$namespaceinfo",
+        this.namespaceResourceSerializer,
+        updatedOperationOptions
+      );
 
-    return this.buildNamespacePropertiesResponse(response);
+      return this.buildNamespacePropertiesResponse(response);
+    } catch (e) {
+      span.setStatus({
```

Comment:
@xirzec I don't see this pattern of updating status on failure in Key Vault. Is this something we added later in the game and missed updating Key Vault?

cc @sadasant

Reply:
BTW, This PR is heavily inspired from storage-blob.